### ISSUE TITLE
V8 WishList: Support $filter=geo.distance(a, b) le distance

### ISSUE THREAD

**Issue by adrianhall** [Azure/azure-mobile-apps#854]
Support distance between two points.

Given a DTO that includes a Microsoft.Spatial.GeographyPoint named "Location", support the following filter:

```text
$filter=geo.distance(location, geo.point(10, 20)) le 1000.00
```

The geo.distance should return a double in kilometers.  Support both 2 and 3 argument forms of geo.point and all the evaluations (eq, ne, le, lt, ge, gt).

**Comment by adrianhall** [comment]
Predecessor: #852

**Comment by adrianhall** [comment]
Fixed in 9b7cd30.  However, this highlighted a client side evaluation bug that needed addressing.  this is now fixed as well.
